pkg/services/secrets/kvstore/plugin_mig.go

```diff
@@ -28,23 +29,23 @@ func ProvidePluginSecretMigrationService(
 	cfg *setting.Cfg,
 	sqlStore sqlstore.Store,
 	secretsService secrets.Service,
-	remoteCheck UseRemoteSecretsPluginCheck,
 	kvstore kvstore.KVStore,
+	manager plugins.SecretsPluginManager,
 ) *PluginSecretMigrationService {
 	return &PluginSecretMigrationService{
 		secretsStore:   secretsStore,
 		cfg:            cfg,
 		logger:         log.New("sec-plugin-mig"),
 		sqlStore:       sqlStore,
 		secretsService: secretsService,
-		remoteCheck:    remoteCheck,
 		kvstore:        kvstore,
+		manager:        manager,
 	}
 }
 
 func (s *PluginSecretMigrationService) Migrate(ctx context.Context) error {
 	// Check if we should migrate to plugin - default false
-	if s.cfg.SectionWithEnvOverrides("secrets").Key("migrate_to_plugin").MustBool(false) && s.remoteCheck.ShouldUseRemoteSecretsPlugin() {
+	if err := shouldUseRemoteSecretsPlugin(s.manager, s.cfg); err == nil {
```

Comment:
we still need to check the `migrate_to_plugin` flag, right?

Reply:
I don't actually remember that conversation, but it makes sense. Maybe before we wanted this to be a flag in case migration is slow, but since we have plans to speed it up, this could be fine.

Reply:
So this means that when use_plugin=true, we will always trigger a migration?
Remember that running the migration means deleting the secrets from the database, so I think that should be explicit to the admin. We would be performing a non-trivial operation.

Reply:
Approved initially but remembered this point and dismissed my review. 

With the PR I'm working on, we'll be migrating to or from the plugin. 

One idea: instead of having `migrate_to_plugin`, we could head this off by having a config setting called `migration_strategy` with the options:
`to_plugin` (default)
`from_plugin`
`none`

This way, by default we have the implied usage, but we give admins the options of keeping old secrets around. Thoughts? @leandro-deveikis @guicaulada

Reply:
@mmandrus just to be sure, in this case we would have a new variable under [secrets] ?

For example:
```
[secrets]
use_plugin = true
migration_strategy = to_plugin
```

Reply:
> One idea: instead of having migrate_to_plugin, we could head this off by having a config setting called [migration_strategy] with the following options mapping to enums:
to_plugin (default)
from_plugin
none

I think that could work, as I said in the chat, we will need some good documentation here

Reply:
@guicaulada that is what I had in mind